MAUI/DataGrid/Row-height-Customization.md

```diff
@@ -0,0 +1,153 @@
+ * `Handled`: This property decides whether the specified height can be set to the row or not. The default value is `false`. When this property is not set, the decided height is not set to the row.
+
+To hook the `SfDatagrid.QueryRowHeight` event, and customize height of a row, follow the code example:
+
+{% highlight c# %}
+//Hooks QueryRowHeight event in SfDataGrid
+dataGrid.QueryRowHeight += DataGrid_QueryRowHeight;
+
+//Event to set the row height on demand
+private void DataGrid_QueryRowHeight(object sender, DataGridQueryRowHeightEventArgs e)
+{
+    //Sets height of the fifth row
+    if (e.RowIndex == 5)
+    {
+        e.Height = 100;
+        e.Handled = true;
+    }
+} 
+{% endhighlight %}
+
+The data grid allows you to customize the height of grid rows with various customizing options while auto calculating the row height based on the content.
+
+## Auto fit the grid rows based on content
+
+The data grid supports the AutoRowHeight feature. The height of a row can be customized based on the content. This can be achieved using the `SfDatagrid.QueryRowHeight` event and the [SfDatagrid.GetIntrinsicRowHeight](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridQueryRowHeightEventArgs.html#Syncfusion_Maui_DataGrid_DataGridQueryRowHeightEventArgs_GetIntrinsicRowHeight_System_Int32_System_Boolean_System_Collections_Generic_List_System_String__) method. The `SfDatagrid.QueryRowHeight` event returns the row height on demand. The `SfDatagrid.GetIntrinsicRowHeight` method returns the height of the row based on the content.
+
+To hook the `SfDatagrid.QueryRowHeight` event and auto fit the height of a row based on content, follow the code example:
+
+{% highlight c# %}
+//Hooks QueryRowHeight event in SfDataGrid to set the row height on demand
+dataGrid.QueryRowHeight += DataGrid_QueryRowHeight;
+
+private void DataGrid_QueryRowHeight(object sender, DataGridQueryRowHeightEventArgs e)
+{
+    if (e.RowIndex != 0)
+    {
+        //Calculates and sets the height of the row based on its content.
+        e.Height = e.GetIntrinsicRowHeight(e.RowIndex);
+        e.Handled = true;
+    }
+}
+{% endhighlight %}
+
+![MAUI DataGrid AutoFit based on row content](Images\row-height-customization\maui-datagrid-autofit.png)
+
+### Calculate height based on certain columns
+
+The data grid allows you to calculate the row height excluding certain columns using the `ExcludeColumns` property.
+
+The following code example illustrates calculating the height of grid rows based on certain columns:
+
+{% highlight c# %}
+ 
+private void DataGrid_QueryRowHeight(object sender, DataGridQueryRowHeightEventArgs e)
+{
+   if(e.RowIndex > 0)
+    {
+        e.Height = e.GetIntrinsicRowHeight(e.RowIndex,false,new List<string> { "Description" });        
+        e.Handled = true;
+    }
+}
+{% endhighlight %}
+
+## Reset row height at runtime
```

Comment:
@NirmalKumarYuvaraj  - Remove this topic.

Reply:
Removed that topic.